pkg/helmer/helmer_test.go

```diff
@@ -0,0 +1,245 @@
+package helmer_test
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-psap/special-resource-operator/pkg/helmer"
+	helmerv1beta1 "github.com/openshift-psap/special-resource-operator/pkg/helmer/api/v1beta1"
+	"github.com/openshift-psap/special-resource-operator/pkg/resource"
+	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/cli"
+	"helm.sh/helm/v3/pkg/repo"
+	v1 "k8s.io/api/core/v1"
+)
+
+const pluginsDir = "../../helm-plugins"
+
+var (
+	ctrl        *gomock.Controller
+	mockCreator *resource.MockCreator
+)
+
+func TestHelmer(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		mockCreator = resource.NewMockCreator(ctrl)
+	})
+
+	RunSpecs(t, "Helmer Suite")
+}
+
+var _ = Describe("helmer_AddorUpdateRepo", func() {
+	It("file:// provider", func() {
+		entry := repo.Entry{
+			Name: "test",
+			URL:  "file://testdata",
+		}
+
+		tempDir := GinkgoT().TempDir()
+
+		repoConfigFile := filepath.Join(tempDir, "config.yaml")
+
+		settings := cli.New()
+
+		settings.PluginsDirectory = pluginsDir
+		settings.RepositoryConfig = repoConfigFile
+		settings.RepositoryCache = filepath.Join(tempDir, "cache")
+
+		err := helmer.NewHelmer(mockCreator, settings).AddorUpdateRepo(&entry)
```

Comment:
Why just we don't mock the whole `Helmer` interface? why are we mocking only the inner `Creator` interface?

Reply:
We cannot mock `Helmer` here, because we are testing its implementation.

Reply:
Ohh Sorry, I though we are testing a higher level.